python/lsst/ts/utils/image_name_service_client.py

```
@@ -0,0 +1,114 @@
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["ImageNameServiceClient"]
+
+import aiohttp
+import json
+import io
+from typing import List
+
+
+class ImageNameServiceClient:
+    def __init__(
+        self,
+        url: str,
+        csc_name: str,
+        csc_index: int,
+        source: str,
+        unit_test: bool = False,
+    ) -> None:
+        """Implement a client for the Image Name Service.
+
+        Parameters
+        ----------
+        url : `str`
+            The image service host.
+            Must be handled by CSC configuration.
+        csc_name : `str`
+            The name of the CSC for matching with the Image Service.
+        csc_index : `int`
+            The index of the CSC, needed for some CSCs which have multiple
+            instances running.
+        source : `str`
+            The two letter ID that the service uses for CSC verification.
+        unit_test : `bool`
+            Is this being used in a unit test?
+
+        Attributes
+        ----------
+        unit_test : `bool`
+            This determines whether to connect to the real service or use
+            fake data.
+        source : `str`
+            The ID used by the service for CSC verification.
+        summit_url : `str`
+            The URL for the summit instance.
+        teststand_url : `str`
+            The URL for the Tucson Test Stand.
+        csc_name : `str`
+            The name of the CSC, used for the mapping of the ID.
+        csc_index : `str`
+            The index of the CSC, used to handle multi instance CSCs.
+
+        """
+        self.unit_test = unit_test
+        self.source = source
+        self.url = url
+        self.csc_name = csc_name
+        self.csc_index = csc_index
+
+    async def get_next_obs_id(self, num_images: int) -> tuple:
+        """Get the observing ID(s).
+
+        Parameters
+        ----------
+        num_images : `int`
+            The number of images to get.
```

What happens if num_images < 1? Please raise an exception and document it in this doc string.